src/matlab2tikz.m

```diff
@@ -3569,8 +3561,20 @@ function warnFacetedInterp(m2t)
                        env, drawOpts, tabOpts, table);
 end
 % ==============================================================================
+function marker = getMarkerInfo(m2t, h)
+    % gets marker-related options as a struct
+    marker                        = struct();
+    marker.style                  = get(h, 'Marker');
+    marker.FaceColor              = get(h, 'MarkerFaceColor');
+    marker.EdgeColor              = get(h, 'MarkerEdgeColor');
+    marker.hasFaceColor           = ~isNone(marker.FaceColor);
+    marker.hasEdgeColor           = ~isNone(marker.EdgeColor);
+    [marker.tikz, marker.options] = translateMarker(m2t, marker.style, ...
+        opts_new(), marker.hasFaceColor);
```

Comment:
Technically, this is not passing a function as an argument. And especially with `opts_*`, you should look at those as objects in an early stage where `opts_new()` is something like `cell()`, `''`, ...

But I think I have an idea to accommodate to your comment: passing that in as a defaulted value should allow to make the code cleaner at the other call site.